docs/source/guides/mod-config.md

```diff
@@ -1,166 +1,323 @@
-	```lua linenums="1"
-	local configPath = "My Awesome Mod" -- The name of the config json file of your mod
-	local defaultConfig = {
-		enabled = true,
-		keybind = {
-			keyCode = tes3.scanCode.k,
-			isShiftDown = false,
-			isAltDown = false,
-			isControlDown = true,
-		}, -- Ctrl + K
-		nonEdible = {
-			["ingred_diamond_01"] = true,
-			["ingred_scrap_metal_01"] = true,
-		}
-	}
+All MCMs will consist of those three types of components. They're building blocks of any mod's config menu. 
+But there are a couple other components that can be useful:
+
+4. **Categories**. These basically work like subsections of a `Page`. 
+	- You can add a `Category` to a page as a way to bundle similar settings together.
+	- You can even make categories inside of other categories.
+5. **Infos**. These can sometimes show up in specific situations as a way to present information that's not directly tied to a specific config setting.
+
+## "Advanced" Examples
 
-	return mwse.loadConfig(configPath, defaultConfig)
-	```
 
-And we import the config with `require` function in the `main.lua`.
 
-!!! example "**MyMod\\main.lua**"
-	```lua linenums="1"
-	local config = require("MyMod.config")
-	```
 
 ### Side Bar Page
 
-MCM has a lot of features, and one of the most commonly used components is the Sidebar Page. When a component is moused over in a sidebar page, the right panel will display the description of that component. It is recommended to use a sidebar page instead of a page when a mod has lots of settings.
+MCM has a lot of features, and one of the most commonly used components is the `SideBarPage`. When a component is moused over in a `SideBarPage`, the right panel will display the `description` of that component. It is recommended to use a `SideBarPage` instead of a regular `Page` whenever a mod has a lot of settings.
 
-!!! example "**MyMod\\main.lua**"
-	```lua
+```lua title="My Awesome Mod/main.lua"
 	local settings = template:createSideBarPage({
 		label = "Settings",
 		-- This description will be displayed at the right panel if no component is moused over.
+		-- (The `\z` character lets you define a string across multiple lines.)
 		description =
-			"My Awesome Mod\n" ..
-			"Version 1.0.0\n" ..
-			"\n" ..
-			"The mod is very awesome. And this is its awesome description.\n",
+			"My Awesome Mod\n\z
+			Version 1.0.0\n\z
+			\n\z
+			The mod is very awesome. And this is its awesome description.\n",
 	})
 
 	settings:createYesNoButton({
 		label = "Enable Mod",
 		-- This description will be displayed at the right panel if the button is moused over.
 		description =
-			"If this setting is enabled, the mod is enabled.\n" ..
-			"\n" ..
-			"If this setting is disabled, the mod is disabled.\n" ..
-			"\n" ..
-			"Default: yes",
+			"If this setting is enabled, the mod is enabled.\n\z
+			\n\z
+			If this setting is disabled, the mod is disabled."
 		variable = mwse.mcm:createTableVariable({ id = "enabled", table = config }),
```

Comment:
Oh no! I tried to get rid all the colons but it looks like I missed this one. I wonder if this guide is the reason so many people use the colon in their MCMs...

But yeah I agree that it would be best to use `configKey` here. I'll make that change.

It might be helpful to explain the `TableVariable` at some point, but I'm not sure of the best place to put that. It just might be confusing if people look at the MCMs of other mods after looking at this guide and see `createTableVariable` everywhere.

Maybe I could just add a `note` or something that explains `TableVariable` and says it's still happening under the hood but for most use-cases the `configKey` should be enough. What do you think?